core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -143,44 +173,62 @@ class SpendReportingService(userInfo: UserInfo, dataSource: SlickDataSource, big
+            .addNamedParameter("projects", stringArrayQueryParameterValue(workspaceProjectsToNames.keySet.map(_.value).toList))
             .build()
 
           result <- bigQueryService.use(_.query(queryJobConfiguration)).unsafeToFuture()
         } yield {
           result.getValues.asScala.toList match {
             case Nil => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"no spend data found for billing project ${billingProjectName.value} between dates $startDate and $endDate"))
-            case rows => extractSpendReportingResults(rows, startDate, endDate)
+            case rows => extractSpendReportingResults(rows, startDate, endDate, workspaceProjectsToNames, aggregationKey)
           }
         }
       }
     }
   }
+
+
+  private def getQuery(aggregationKey: SpendReportingAggregationKey, tableName: String): String = {
+    aggregationKey match {
+      case SpendReportingAggregationKeys.Daily => s"""
+                                                     | SELECT
+                                                     |  SUM(cost) as cost,
+                                                     |  SUM(IFNULL((SELECT SUM(c.amount) FROM UNNEST(credits) c), 0)) as credits,
+                                                     |  currency,
+                                                     |  DATE(_PARTITIONTIME) as date
+                                                     | FROM `$tableName`
+                                                     | WHERE billing_account_id = @billingAccountId
+                                                     | AND _PARTITIONTIME BETWEEN @startDate AND @endDate
+                                                     | AND project.id in UNNEST(@projects)
+                                                     | GROUP BY currency, date
+                                                     |""".stripMargin
+      case SpendReportingAggregationKeys.Workspace => s"""
+                                                         | SELECT
+                                                         |  SUM(cost) as cost,
+                                                         |  SUM(IFNULL((SELECT SUM(c.amount) FROM UNNEST(credits) c), 0)) as credits,
+                                                         |  currency,
+                                                         |  project.id as googleProjectId
+                                                         | FROM `$tableName`
+                                                         | WHERE billing_account_id = @billingAccountId
+                                                         | AND _PARTITIONTIME BETWEEN @startDate AND @endDate
+                                                         | AND project.id in UNNEST(@projects)
+                                                         | GROUP BY currency, googleProjectId
+                                                         |""".stripMargin
```

Comment:
It seems like the only difference here is the Group By parameter. Would it make sense to have the string as a variable (plugging in the parameter) to avoid duplication and ensure that they queries do not get out of sync?

Reply:
Thanks for making this change!